be/main/task_views.py

```diff
@@ -15,18 +15,47 @@
 
 from latex import build_pdf
 
+from integrations.credit_kudos.api import (
+    get_access_token,
+    get_inflows_over_time,
+    get_latest_report,
+)
+from reports.models import IncomeReport
+
 logger = logging.getLogger(__name__)
 
 
 def create_pdf(request):
     template = loader.get_template("pdf.tex")
-    context = json.loads(request.body)
-    context["logo_path"] = os.path.join(settings.BASE_DIR, "static/images/creditkudos.png")
+    params = json.loads(request.body)
+    income_report_id = params["income_report_id"]
+    income_report = IncomeReport.objects.get(
+      id=income_report_id,
+    )
+
+    report = get_latest_report(income_report)
+    if report["status"] != "complete":
+      # Report not ready yet.
+      return HttpResponse(status=422)
```

Comment:
```suggestion
    if report["status"] != "complete":
        # Report not ready yet.
        return HttpResponse(status=422)
```

Python needs 4 spaces for indents